lib/src/rules/use_super_initializers.dart

```diff
@@ -0,0 +1,247 @@
+  static const LintCode singleParam =
+      LintCode('use_super_initializers', "Convert '{0}' to a super parameter.");
+  static const LintCode multipleParams =
+      LintCode('use_super_initializers', 'Convert {0} to super parameters.');
+
+  UseSuperInitializers()
+      : super(
+            name: 'use_super_initializers',
+            description: _desc,
+            details: _details,
+            maturity: Maturity.experimental,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    if (!context.isEnabled(Feature.super_parameters)) return;
+
+    var visitor = _Visitor(this, context);
+    registry.addConstructorDeclaration(this, visitor);
+  }
+
+  @visibleForTesting
+  static String joinIdentifiers(List<String> identifiers) =>
+      identifiers.quotedAndCommaSeparatedWithAnd;
+}
+
+class _Visitor extends SimpleAstVisitor {
+  final LinterContext context;
+  final LintRule rule;
+
+  _Visitor(this.rule, this.context);
+
+  void check(SuperConstructorInvocation superInvocation,
+      FormalParameterList parameters) {
+    var constructorElement = superInvocation.staticElement;
+    if (constructorElement == null) return;
+
+    // todo(pq): consolidate logic shared w/ server
+    //  (https://github.com/dart-lang/linter/issues/3263)
+
+    // Bail if there are positional params that can't be converted.
+    if (!_checkForConvertiblePositionalParams(
+        constructorElement, superInvocation, parameters)) {
+      return;
+    }
+
+    for (var parameter in parameters.parameters) {
+      var parameterElement = parameter.declaredElement;
+      if (parameterElement == null) continue;
+      if (parameterElement.isNamed) {
+        if (_checkNamedParameter(
+            parameter, parameterElement, constructorElement, superInvocation)) {
+          _reportOnIdentifier(parameter.identifier);
+        }
+      }
+    }
+  }
+
+  @override
+  visitConstructorDeclaration(ConstructorDeclaration node) {
+    for (var initializer in node.initializers.reversed) {
+      if (initializer is SuperConstructorInvocation) {
+        check(initializer, node.parameters);
+        return;
+      }
+    }
+  }
+
+  /// Check if all super positional parameters can be converted to use super-
+  /// initializers. Return `false` if there are parameters that can't be
+  /// converted since this will short-circuit the lint.
+  bool _checkForConvertiblePositionalParams(
+      ConstructorElement constructorElement,
+      SuperConstructorInvocation superInvocation,
+      FormalParameterList parameters) {
+    var positionalSuperArgs = superInvocation.argumentList.arguments
+        .whereType<SimpleIdentifier>()
+        .toList();
+    if (positionalSuperArgs.isEmpty) return true;
+
+    var constructorPositionalParams =
+        constructorElement.parameters.where((p) => !p.isNamed).toList();
+    if (positionalSuperArgs.length < constructorPositionalParams.length) {
```

Comment:
I don't think this is the right condition because the `constructorPositionalParams` might include optional positional parameters that aren't given in the super invocation:
```
class A {
  A(int a, [int b]);
}
class B extends A {
  B(int x) : super(x); // LINT
}
```

`positionalSuperArgs.length` == 1
`constructorPositionalParams.length` == 2

Also consider:
```
class A {
  A(int x, int y, [int z]);
}
class B extends A {
  B(int a, int b) : super(a, 2, b); // OK
}
```

This shouldn't produce a lint because there's an argument that isn't a simple identifier.

I don't think the number of `constructorPositionalParams` matters at all. I think the only requirements are that
- all of the positional arguments are simple identifiers that resolve to a parameter from the constructor being linted, and
- the ones you're checking starting on line 138.

Reply:
Excellent.  Thanks!